nupic/research/spatial_pooler.py

```diff
@@ -965,9 +964,10 @@ def _adaptSynapses(self, inputVector, activeColumns):
     permChanges.fill(-1 * self._synPermInactiveDec)
     permChanges[inputIndices] = self._synPermActiveInc
     for i in activeColumns:
-      perm = self._permanences.getRow(i)
-      maskPotential = numpy.where(self._potentialPools.getRow(i) > 0)[0]
-      perm[maskPotential] += permChanges[maskPotential]
+      synapses = self.connections.synapsesForSegment(i)
+      inputIndices = self._getInputIndexesFromSynapses(synapses)
+      perm = self._getPermanencesFromColumn(i)
+      perm += permChanges[inputIndices]
```

Comment:
some trouble in adaptSynapses() and the failing test https://travis-ci.org/numenta/nupic/jobs/59624260#L3493